lib/slave/src/AbstractESC.cc

```diff
@@ -26,12 +26,35 @@ namespace kickcat
 
         bool is_valid = (sm_read.start_address == sm_ref.start_address) and
                         (sm_read.length == sm_ref.length) and
-                        (sm_read.control == sm_ref.control);
+                        (sm_read.control == sm_ref.control) and
+                        sm_read.activate == 1;
 
-        printf("SM read %i: start address %x, length %u, control %x, status %x, activate %x \n", sm_ref.index, sm_read.start_address, sm_read.length, sm_read.control, sm_read.status, sm_read.activate);
+        // printf("SM read %i: start address %x, length %u, control %x, status %x, activate %x \n", sm_ref.index, sm_read.start_address, sm_read.length, sm_read.control, sm_read.status, sm_read.activate);
+        // printf("SM config %i: start address %x, length %u, control %x \n", sm_ref.index, sm_ref.start_address, sm_ref.length, sm_ref.control);
         return is_valid;
     }
 
+    void AbstractESC::set_sm_activate(SyncManagerConfig const& sm_conf, bool is_activated)
+    {
+        auto create_sm_address = [](uint16_t reg, uint16_t sm_index)
+        {
+            return reg + sm_index * 8;
+        };
+
+        SyncManager sm;
+        reportError(read(create_sm_address(reg::SYNC_MANAGER, sm_conf.index), &sm, sizeof(sm)));
```

Comment:
no return ? (instead of just a trace)

Reply:
In the whole class the errors are just traced, I didn't use them to do meaningful things